### DISCUSSION TITLE
Would it be correct to implement k-gradient step updates like this?

### DISCUSSION THREAD

**Discussion by AhmetHamzaEmra** [google/flax#2799] [Q&A] [answered]
```
@jax.jit
def update_inner(model, data, other_data, z):
    def loss_fn(params):
        logits = Net().apply({'params': params}, data['kip'])
        loss = aug_cross_entropy_loss(logits=logits, labels= data['label'], z=z)
        accuracy = jnp.mean(jnp.argmax(logits, -1) == data['label'])
        return loss, accuracy

    grad_fn = jax.value_and_grad(loss_fn, has_aux=True)
    for _ in range(k_steps): # user provided parameter
        aux, grads = grad_fn(model.params)
        loss, acc = aux
        model = model.apply_gradients(grads=grads)
# Some other code later
```

Thanks

**Comment by marcvanzee** [comment, marked as answer]
Looks fine to me, assuming `model` is a `TrainState`-like object: https://github.com/google/flax/blob/33afdc9f8638fbd314579c6c4c6c0d253ed45c3e/flax/training/train_state.py#L22

**Reply by AhmetHamzaEmra** [reply]
thanks
